shoes-core/lib/shoes/http_wrapper.rb

```diff
@@ -0,0 +1,55 @@
+require 'uri'
+require 'net/http'
+
+# Wrapper for Net::HTTP to provide us to the functionality that we need which
+# isn't easily available from any particular gem out there :(
+#
+# Specific things this class does that we need include:
+#
+#   1) Reading in chunks to trigger progress
+#   2) Allow for graceful redirection
+#
+# Most of the gems out there do 2) but not 1), or do both but are untested on
+# JRuby.
+#
+class Shoes
+  class HttpWrapper
+    def execute(url, meth, body, headers = {}, redirects_left = 5, &blk)
```

Comment:
Funny to hear you mirror things I thought about precisely while doing the work ;)

An earlier rev of this had state with the object, so it started life as a true object. I left it when I considered a class method because it "might" want state, but after thinking a little longer, I think it's be better as a class method. Will revise.